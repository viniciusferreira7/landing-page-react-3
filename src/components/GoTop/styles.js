import styled, { css } from 'styled-components';

export const Container = styled.a`
${({ theme }) => css`
  background-color:${theme.colors.primaryColor};
  color:${theme.colors.white};
  display: flex;
  align-items: center;
  justify-content: center;
  position: fixed;
  bottom: 2rem;
  right: 2rem;
  width:4rem;
  height:4rem;
  border-radius: 2rem;
  z-index: 6;
  cursor: pointer;
  opacity: 0.7;
`}
`;
