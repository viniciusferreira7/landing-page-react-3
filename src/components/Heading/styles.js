import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) =>
    css`
      font-size: ${theme.font.sizes.medium};
    `,
  medium: (theme) =>
    css`
      font-size: ${theme.font.sizes.large};
    `,
  big: (theme) =>
    css`
      font-size: ${theme.font.sizes.xlarge};
    `,
  huge: (theme) =>
    css`
      font-size: ${theme.font.sizes.xhuge};
      ${mediaFont(theme)}
    `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.xlarge};
  }
`;

const titleCase = (upperCase) => css`
  text-transform: ${upperCase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, upperCase }) => css`
    color: ${!colorDark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)};
    ${titleCase(upperCase)}
  `}
`;
