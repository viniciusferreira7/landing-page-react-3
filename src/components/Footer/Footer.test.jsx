import { screen } from '@testing-library/react';
import { Footer } from '.';
import { renderTheme } from '../../styles/render-theme';

describe('<Footer/>', () => {
  it('should render', () => {
    const { container } = renderTheme(<Footer html="<h1>Texto</h1>" />);
    expect(screen.getByRole('heading', { name: 'Texto' }));
    expect(container).toMatchSnapshot();
  });
});
