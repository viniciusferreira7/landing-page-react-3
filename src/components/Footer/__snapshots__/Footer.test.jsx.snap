// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> should render 1`] = `
.c0 {
  text-align: center;
}

.c0 a {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
}

.c2 {
  font-size: 2.4rem;
}

.c1 {
  width: 100%;
  margin: 0 auto;
  padding: 3.2rem;
}

<div>
  <div
    class="c0"
  >
    <section
      class="c1"
    >
      <div
        class="c2"
      >
        <h1>
          Texto
        </h1>
      </div>
    </section>
  </div>
</div>
`;
